src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs

```diff
@@ -35,6 +35,15 @@ internal class CSharpRecommendationService : AbstractRecommendationService
             var hideAdvancedMembers = options.GetOption(RecommendationOptions.HideAdvancedMembers, semanticModel.Language);
             symbols = symbols.FilterToVisibleAndBrowsableSymbols(hideAdvancedMembers, semanticModel.Compilation);
 
+            var localdeclarationSyntax = context.TargetToken.Parent?.GetAncestor<VariableDeclaratorSyntax>();
+            if (localdeclarationSyntax != null && 
+                (localdeclarationSyntax.GetAncestor<VariableDeclarationSyntax>().Type.IsVar || 
+                localdeclarationSyntax.DescendantNodes().Any(n => n.IsKind(SyntaxKind.CastExpression))))
```

Comment:
i really don't like broad checks like this.  They seem like they coudl fire far too often (for example, if you have a nested lambda with a cast anywhere within it).

Reply:
Given the bug this is fixing.  It seems like it would be much easier to see if you were on the right of a cast expression, and just remove the 'cast' type from the list if that's the case.  No need to do anything replaced to variable-declarators/var.

Reply:
@CyrusNajmabadi ```int a = a = 1;``` is legal syntax but ```var a = a = 1;``` is not legal syntax.

```csharp
interface IVsSolution {}
interface IVsSolution5{}

class C
{
    void M()
    {
        IVsSolution5 solution = null;
        var solution5 = (IVsSolution5)solution;
    }
}
```

It is also legal syntax, I don't think that ```solution``` should be removed.

Reply:
Need to take into account and it is reasonable to provided unit tests for 
1. Lambdas and local functions
2.  tuples var (solution5, solution) = ((IVsSolution5)solution, solution$$)
3. parameters
var a = method((IVsSolution5)solution, solution$$)

---
In reply to: [156197458](https://github.com/dotnet/roslyn/pull/23696#discussion_r156197458) [](ancestors = 156197458)